### ISSUE TITLE
zuletzt vom Adapter registrierte SpotArea in einem Datenpunkt festhalten

### ISSUE THREAD

**Issue by chris76e** [mrbungle64/ioBroker.ecovacs-deebot#132]
Hi,

möchte gerne in meiner VIS die letzten gereinigten Berreich anzeigen lassen.
Wäre toll wenn man dafür einen Datenpunkt anlegen könnte.

Danke

Christian

Deebot 900

**Comment by mrbungle64** [comment]
@chris76e 

Sehe ich das richtig, dass das eigentlich das gleiche wäre wie:

* map.deebotPositionCurrentSpotAreaID
* map.deebotPositionCurrentSpotAreaName

Nur mit dem Unterschied, dass die Werte nicht geleert werden und so jeweils der letzte stehen bleibt?

**Comment by chris76e** [comment]
> 
> 
> @chris76e
> 
> Sehe ich das richtig, dass das eigentlich das Gleiche wäre wie:
> 
>     * map.deebotPositionCurrentSpotAreaID
> 
>     * map.deebotPositionCurrentSpotAreaName
> 
> 
> Nur mit dem Unterschied, dass die Werte nicht geleert werden und so jeweils der letzte Wert stehen bleibt?

@mrbungle64 

das meinte ich nicht, da steht ja eigentlich immer als letztes die SpotArea in dem die Ladestation steht.
Habe bei mir in der VIS verschieden buttons wo mit ich direkt eine SpotArea auswählen kann. Und dies sollte dann als Datenpunkt angelegt werden.

**Comment by mrbungle64** [comment]
> > @chris76e
> > Sehe ich das richtig, dass das eigentlich das Gleiche wäre wie:
> > ```
> > * map.deebotPositionCurrentSpotAreaID
> > 
> > * map.deebotPositionCurrentSpotAreaName
> > ```
> > 
> > 
> > Nur mit dem Unterschied, dass die Werte nicht geleert werden und so jeweils der letzte Wert stehen bleibt?
> 
> @mrbungle64
> 
> das meinte ich nicht, da steht ja eigentlich immer als letztes die SpotArea in dem die Ladestation steht.
> Habe bei mir in der VIS verschieden buttons wo mit ich direkt eine SpotArea auswählen kann. Und dies sollte dann als Datenpunkt angelegt werden.

Ja, das solltest Du mir ja mal beschreiben, was da für Dich der Use Case ist 😉 
Ist aber auch schwierig, weil es auch den Datenpunkt "control.spotArea" gibt, wo man Komma-separiert mehrere SpotAreas eintragen kann. Da ist dann aber nicht sichergestellt, dass der Saugroboter das auch in genau dieser Reihenfolge macht.

**Comment by mrbungle64** [comment]
@chris76e 

Die einzige vernünftige Lösung war dann den letzten Wert aus "map.deebotPositionCurrentSpotAreaID" (bzw. Name) festzuhalten, während "status.device" `cleaning` ist.

**Comment by chris76e** [comment]
> Btw.: Wenn Du das genau für diese Buttons in Deiner VIS haben möchtest, geht das bestimmt recht bequem per Skript.

Habe ich mir auch schon gedacht. Werde es dann mal als script versuchen.

**Comment by mrbungle64** [comment]
Ich schließe den Issue mal, da hier von meiner Seite aus nichts offen war.

**Comment by mrbungle64** [comment]
https://forum.iobroker.net/post/914327

**Comment by mrbungle64** [comment]
@chris76e 

Die Funktion ist in der 1.4.10-alpha.0 enthalten.

Es gibt es nun den Kanal "map.lastCleanedSpotArea"

Folgende Datenpunkte sind aktuell vorhanden:

"map.lastCleanedSpotArea.mapID": ID der Map (Interessant bei mehreren Karten)
"map.lastCleanedSpotArea.spotAreaID": ID der Spot Area
"map.lastCleanedSpotArea.spotAreaName": Name des Bereichs (Spot Area)
"map.lastCleanedSpotArea.totalSeconds": Dauer der Reinigung in Sekunden
"map.lastCleanedSpotArea.totalTime": Dauer der Reinigung im Format "h mm ss" (z.B. "0h 03m 17s")
